### PR TITLE
Enable nullable reference types by default 

### PR DESCRIPTION
**Pull request by drewnoakes**
### Context

In https://github.com/dotnet/msbuild/pull/7093#discussion_r759775351 we observed that adding `#nullable enable` would better document the intended nullness of values.

It would be good if new code was `#nullable enable` by default.

### Changes Made

This turns C# 9's nullable reference types on by default for all projects in the solution, and opts out unannotated files by adding `#nullable disable` to them.

This has two nice properties:

1. New code will be written with nullability in mind as the feature will be on by default.
2. It's easy to find unannotated code by searching for `#nullable disable`.

### Testing

CI. The nullable reference types feature does not impact code at runtime, other than the addition of metadata.

### Notes

We have taken this approach in the .NET Project System and in CPS, and it's worked very well for us.

Code under `Deprecated` is unchanged.

This is the kind of change a team should discuss first. I was curious if it'd be an easy change for MSBuild and by the time I had an answer, I also had a PR. I will not be offended if this is not a change you wish to make at this time.

### CODE REVIEW

**Comment by rainersigwald** [comment]
I think I'm on board with this? I wasn't willing to do it myself and I still find the approach kinda ugly, but it does encourage modernization and make the old-and-busted unannotated files stand out. And I don't have a better approach--without a mechanism to enforce "new files get nullable enabled by default/if you do major surgery in a file enable it" we were just kinda sitting still.

**Comment by rainersigwald** [comment]
> I made this change with a regex, matching on `^namespace `.

I [learned this morning](https://devblogs.microsoft.com/visualstudio/whats-new-in-net-productivity-with-visual-studio-2022/) that there's a VS refactoring to do this!

We discussed this with the team and we're planning to go ahead with this approach--thanks for piloting it out for us!

File: src/Shared/ExceptionHandling.cs [hunk 1]
```diff
@@ -2,6 +2,8 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 #if BUILDINGAPPXTASKS
+#nullable disable
+
```
Comment (rainersigwald) [inline src/Shared/ExceptionHandling.cs]: Just move this above the `#if`s?

**Review by rainersigwald — APPROVED** [review]

File: src/Shared/ExceptionHandling.cs (hunk 1 above)
Comment (drewnoakes) [reply src/Shared/ExceptionHandling.cs:5]: Fixed. Merged latest main and resolved conflict.

**Pull request merged**